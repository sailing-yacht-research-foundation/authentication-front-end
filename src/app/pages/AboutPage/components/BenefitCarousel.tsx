import '../styles/benefit-carousel.css';
import React from 'react';
import { Carousel, Col, Row } from 'antd';
import styled from 'styled-components';
import { media } from 'styles/media';

import PasswordBg from '../assets/Image-1.svg';
import AccessControlBg from '../assets/Image-2.svg';
import OrganizationBg from '../assets/Image-3.svg';
import GDPRBg from '../assets/Image-4.svg';
import { useTranslation } from 'react-i18next';
import { translations } from 'locales/translations';

const backgroundProperties = {
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'contain',
    backgroundPosition: '100% -50%'
}

export const BenefitCarousel = (props) => {

    const { t } = useTranslation();

    return (
        <Carousel
            slidesToShow={1}
            dots={{ className: 'carousel-dot custom' }}>
            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={24} xs={24} sm={24}>
                            <CarouselText className="section-header-text">{t(translations.about_page.benefit_carousel.slide_one.title)}</CarouselText>
                            <CarouselItemDescription>{t(translations.about_page.benefit_carousel.slide_one.description)}</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={24} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${PasswordBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">{t(translations.about_page.benefit_carousel.slide_two.title)}</CarouselText>
                            <CarouselItemDescription>{t(translations.about_page.benefit_carousel.slide_two.description)}</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${AccessControlBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">{t(translations.about_page.benefit_carousel.slide_three.title)}</CarouselText>
                            <CarouselItemDescription>{t(translations.about_page.benefit_carousel.slide_three.description)}</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${GDPRBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">{t(translations.about_page.benefit_carousel.slide_four.title)}</CarouselText>
                            <CarouselItemDescription>{t(translations.about_page.benefit_carousel.slide_four.description)}</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${OrganizationBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>
        </Carousel>
    )
}

const CarouselOuterContainer = styled.div`
`;

const CarouselInnerContainer = styled.div`
    padding: 0 120px;
    padding-bottom: 80px;
`;

const CarouselImage = styled.div`
    height: 100%;
    background-size: cover;
    width: 100%;
    margin-top: 40px;
    height: 420px;

    ${media.medium`
        margin-top: 40px;
        height: 420px;
        background-position: 100% -50% !important;
    `};   
    
    ${media.small`
        margin-top: 40px;
        height: 420px;
        background-position: 50% 50% !important;
    `};  

    ${media.large`
        margin-top: 0 !important;
        height: 500px !important;
        background-position: 100% -50% !important;
    `};   
`;

const CarouselText = styled.h1`
    max-width: 499px;
    margin-top: 88px;
    text-align: center;

    ${media.large`
        text-align: left;
    `};

    ${media.medium`
        text-align: left;
    `};

    ${media.small`
        text-align: left;
    `};
`;

const CarouselItemDescription = styled.p`
    text-align: left;
    font-size: 14px;
    font-weight: 400px;
    line-height: 28px;
    color: #737475;

    text-align: center;

    ${media.large`
        text-align: left;
    `};

    ${media.medium`
        text-align: left;
    `};

    ${media.small`
        text-align: left;
    `};
`;