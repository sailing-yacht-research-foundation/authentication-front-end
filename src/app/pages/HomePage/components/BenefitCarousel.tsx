import '../styles/benefit-carousel.css';
import React from 'react';
import { Carousel, Col, Row } from 'antd';
import styled from 'styled-components';
import { media } from 'styles/media';

import PasswordBg from '../assets/one-password.svg';
import AccessControlBg from '../assets/access-control.svg';
import OrganizationBg from '../assets/organization.svg';
import GDPRBg from '../assets/gdpr.svg';

const backgroundProperties = {
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'contain',
    backgroundPosition: '100% -50%'
}

export const BenefitCarousel = (props) => {

    return (
        <Carousel
            slidesToShow={1}
            dots={{ className: 'carousel-dot custom' }}>
            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={24} xs={24} sm={24}>
                            <CarouselText className="section-header-text">One password for all sailing app and data</CarouselText>
                            <CarouselItemDescription>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={24} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${PasswordBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">You control who can access your sailing data</CarouselText>
                            <CarouselItemDescription>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${AccessControlBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">GDPR & CCPA Compliant</CarouselText>
                            <CarouselItemDescription>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${GDPRBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>

            <CarouselOuterContainer>
                <CarouselInnerContainer className="carousel-inner-container">
                    <Row className="carousel-row">
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselText className="section-header-text">Organization independent</CarouselText>
                            <CarouselItemDescription>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum</CarouselItemDescription>
                        </Col>
                        <Col lg={12} md={12} xs={24} sm={24}>
                            <CarouselImage className="carousel-image-container" style={{ backgroundImage: `url(${OrganizationBg})`, ...backgroundProperties }} />
                        </Col>
                    </Row>
                </CarouselInnerContainer>
            </CarouselOuterContainer>
        </Carousel>
    )
}

const CarouselOuterContainer = styled.div`
`;

const CarouselInnerContainer = styled.div`
    padding: 0 120px;
    padding-bottom: 80px;
`;

const CarouselImage = styled.div`
    height: 100%;
    background-size: cover;
    width: 100%;
    margin-top: 40px;
    height: 420px;

    ${media.medium`
        margin-top: 40px;
        height: 420px;
        background-position: 100% -50% !important;
    `};   
    
    ${media.small`
        margin-top: 40px;
        height: 420px;
        background-position: 50% 50% !important;
    `};  

    ${media.large`
        margin-top: 0 !important;
        height: 500px !important;
        background-position: 100% -50% !important;
    `};   
`;

const CarouselText = styled.h1`
    max-width: 499px;
    margin-top: 88px;
    text-align: center;

    ${media.large`
        text-align: left;
    `};

    ${media.medium`
        text-align: left;
    `};

    ${media.small`
        text-align: left;
    `};
`;

const CarouselItemDescription = styled.p`
    text-align: left;
    font-size: 14px;
    font-weight: 400px;
    line-height: 28px;
    color: #737475;

    text-align: center;

    ${media.large`
        text-align: left;
    `};

    ${media.medium`
        text-align: left;
    `};

    ${media.small`
        text-align: left;
    `};
`;