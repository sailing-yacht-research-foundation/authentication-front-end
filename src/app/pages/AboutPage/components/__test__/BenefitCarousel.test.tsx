import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { BenefitCarousel } from '../BenefitCarousel';

const shallowRenderer = createRenderer();

describe('<BenefitCarousel />', () => {
  it('should render the benefit carousel and matchs the snapshot', () => {
    shallowRenderer.render(<BenefitCarousel />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});