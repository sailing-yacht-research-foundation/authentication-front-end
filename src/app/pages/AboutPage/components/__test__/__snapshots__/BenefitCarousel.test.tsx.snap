// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BenefitCarousel /> should render the benefit carousel and matchs the snapshot 1`] = `
<ForwardRef
  dots={
    Object {
      "className": "carousel-dot custom",
    }
  }
  slidesToShow={1}
>
  <styled.div>
    <styled.div
      className="carousel-inner-container"
    >
      <Row
        className="carousel-row"
      >
        <Col
          lg={12}
          md={24}
          sm={24}
          xs={24}
        >
          <styled.h1
            className="section-header-text"
          >
            One password for all sailing apps and data:
          </styled.h1>
          <styled.p>
            syrf.io allows third party developers to use your SYRF profile for authentication. This means you have one less password to remember for your favorite sailing apps.
          </styled.p>
        </Col>
        <Col
          lg={12}
          md={24}
          sm={24}
          xs={24}
        >
          <styled.div
            className="carousel-image-container"
            style={
              Object {
                "backgroundImage": "url(Image-1.svg)",
                "backgroundPosition": "100% -50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
              }
            }
          />
        </Col>
      </Row>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      className="carousel-inner-container"
    >
      <Row
        className="carousel-row"
      >
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.h1
            className="section-header-text"
          >
            You control who can access your sailing data:
          </styled.h1>
          <styled.p>
            syrf.io is like Google Drive for all of your sailing data. 3rd party apps (such as those from your favorite sailing organization) can request to import your data. You retain strict control over who does and doesnâ€™t have access.
          </styled.p>
        </Col>
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.div
            className="carousel-image-container"
            style={
              Object {
                "backgroundImage": "url(Image-2.svg)",
                "backgroundPosition": "100% -50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
              }
            }
          />
        </Col>
      </Row>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      className="carousel-inner-container"
    >
      <Row
        className="carousel-row"
      >
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.h1
            className="section-header-text"
          >
            GDPR & CCPA Compliant:
          </styled.h1>
          <styled.p>
            We take your privacy seriously. Our apps go above and beyond to ensure we only collect race related data and that your data is anonymized and protected. We never, EVER, sell your data to 3rd parties and our revenue model does not rely on personalized advertising. We only use cookies with your permission and you are free to delete your data at any time.
          </styled.p>
        </Col>
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.div
            className="carousel-image-container"
            style={
              Object {
                "backgroundImage": "url(Image-4.svg)",
                "backgroundPosition": "100% -50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
              }
            }
          />
        </Col>
      </Row>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      className="carousel-inner-container"
    >
      <Row
        className="carousel-row"
      >
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.h1
            className="section-header-text"
          >
            Organization independent:
          </styled.h1>
          <styled.p>
            The sailing world is full of passionate groups of people. We decided to build tech that enables everyone, rather than hitching our wagon to any one organization in particular. By staying unaffiliated, SYRF is able to stay innovative and build tools for everyone.
          </styled.p>
        </Col>
        <Col
          lg={12}
          md={12}
          sm={24}
          xs={24}
        >
          <styled.div
            className="carousel-image-container"
            style={
              Object {
                "backgroundImage": "url(Image-3.svg)",
                "backgroundPosition": "100% -50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
              }
            }
          />
        </Col>
      </Row>
    </styled.div>
  </styled.div>
</ForwardRef>
`;
